python/lsst/afw/detection/bootprint.cc

```diff
@@ -0,0 +1,123 @@
+/*
+ * LSST Data Management System
+ * Copyright 2008-2017  AURA/LSST.
+ *
+ * This product includes software developed by the
+ * LSST Project (http://www.lsst.org/).
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the LSST License Statement and
+ * the GNU General Public License along with this program.  If not,
+ * see <https://www.lsstcorp.org/LegalNotices/>.
+ */
+
+
+#include <cstdint>
+#include <pybind11/pybind11.h>
+#include <pybind11/stl.h>
+#include "numpy/arrayobject.h"
+#include "ndarray/pybind11.h"
+#include "ndarray/converter.h"
+#include "lsst/afw/table/io/Persistable.h"
+#include "lsst/afw/table/io/pybind11.h" // for declarePersistableFacade
+#include "lsst/afw/detection/Bootprint.h"
+
+namespace py = pybind11;
+using namespace pybind11::literals;
+
+namespace lsst { namespace afw { namespace detection {
+
+PYBIND11_PLUGIN(_bootprint) {
+    py::module mod("_bootprint", "Python wrapper for afw Bootprint library");
+    table::io::declarePersistableFacade<Bootprint>(mod, "Bootprint");
+
+    if (_import_array() < 0) {
+        PyErr_SetString(PyExc_ImportError, "numpy.core.multiarray failed to import");
+        return nullptr;
+    }
+
+    /* Bootprint Constructors */
+    py::class_<Bootprint,
+               std::shared_ptr<Bootprint>,
+               daf::base::Citizen,
+               table::io::Persistable,
+               table::io::PersistableFacade<Bootprint>> clsBootprint(mod, "Bootprint");
+    clsBootprint.def(py::init<std::shared_ptr<geom::SpanSet>, geom::Box2I const &>(),
+                     "inputSpans"_a, "region"_a = geom::Box2I());
+
+    clsBootprint.def(py::init<std::shared_ptr<geom::SpanSet>,
+                              afw::table::Schema const &,
+                              geom::Box2I const &>(),
+                     "inputSpans"_a, "peakSchema"_a, "region"_a = geom::Box2I());
+
+    /* Bootprint Methods */
+    clsBootprint.def("getSpans", &Bootprint::getSpans);
+    clsBootprint.def("setSpans", &Bootprint::setSpans);
+    clsBootprint.def("getPeaks", (PeakCatalog & (Bootprint::*)()) &Bootprint::getPeaks,
+                     py::return_value_policy::reference);
```

Comment:
Are you sure? What happens if the `Bootprint` object is destroyed before the return value is? Perhaps you want `reference_internal`?